drivers/ethernet/eth_nxp_enet.c

```diff
@@ -308,6 +315,14 @@ static enum ethernet_hw_caps eth_nxp_enet_get_capabilities(const struct device *
 		ETHERNET_HW_RX_CHKSUM_OFFLOAD |
 #endif
 		ETHERNET_LINK_100BASE_T;
+
+#ifdef ENET_HAS_AVB
+	if (config->phy_mode == NXP_ENET_RGMII_MODE) {
+		caps |= ETHERNET_LINK_1000BASE_T;
+	}
+#endif /* ENET_HAS_AVB */
```

Comment:
why is it guarded by ENET_HAS_AVB?

Reply:
`ENET_HAS_AVB` is defined if `defined(FSL_FEATURE_ENET_HAS_AVB) && FSL_FEATURE_ENET_HAS_AVB`.
`FSL_FEATURE_ENET_HAS_AVB` is used by the underlying SDK ENET driver to determine if it should define  `kENET_RgmiiMode`, `kENET_MiiSpeed1000M` etc.

It could use `#if defined(ENET_1G)` instead, but I am using it here for consistency with the SDK driver.
According to user manual, the presence of Audio Video Bridging implies ENET_1G:

```
ENET supports 10M/100M Ethernet, and has the following key features:
• 3.3V RMII and MII operation
• IEEE1588

ENET_1G supports 1G Ethernet, and has the following features:
• RGMII/RMII/MII operation
• IEEE1588
• AVB
```

Reply:
There should probably be a kconfig indicating ENET1G support, and we would use that instead

Reply:
I will try to improve this and add KConfig option.

Reply:
Hi @decsny,

So I was thinking about adding `HAS_MCUX_ENET1G` into `modules/Kconfig.mcux` as there is also `HAS_MCUX_ENET` (which I am not sure if it is used anywhere).
Then there is `ETH_NXP_ENET` in `drivers/ethernet/Kconfig.nxp_enet`. `CONFIG_ETH_NXP_ENET` is used from clock control and SoC code. I am not sure if adding `ETH_NXP_ENET1G` is ok, as both share the same driver.
Then there are definitions like `config ETH_NXP_ENET_HW_ACCELERATION`.

So we have about four options:

- Using `CONFIG_HAS_MCUX_ENET1G`.
- Using `CONFIG_ETH_NXP_ENET1G`.
- Using `CONFIG_ETH_NXP_ENET_ENET1G` (or similar name) as a subconfig.
- Or asking device tree if `ENET1G` is enabled.

Which one is preferred? Or something different? Thanks.

Reply:
Should not add HAS_MCUX_ENET1G, those kconfigs in that file we are trying to deprecate, as I think they were from before we used devicetree to determine if a platform has a device.

I think the name of the kconfig does not matter so much, but my first instinct is that it should be a dependency of the driver kconfig. ie the kconfig file for the driver would look basically like:
```
config ETH_NXP_ENET
...

if ETH_NXP_ENET

config ETH_NXP_ENET1G (or whatever name)
bool
default y
depends on DT_HAS_NXP_ENET_1G_ENABLED
....

endif
```

and therefore yes, I am thinking it might be correct to add another compatible to DT for this purpose. i.e., the node would have 2 compatibles, "nxp,enet" and "nxp,enet1g". Same driver but the 2nd compatible indicates the variation of the device. I will admit I haven't actually done this before but I would like to experiment with this idea to see how it works, you will probably need to add like an empty binding so that the DT tooling makes the kconfig and macros for it.

The reason I say this is because then the code can be conditionally included without user needing to manually select the kconfig, DT is basically THE standard way for us to automatically configure the software based on the platform needs, so it makes sense to me to attempt this sort of approach.

Reply:
and also, I think you could even add potentially a static assertion (using BUILD_ASSERT macro) in this driver device init macro that checks that in the case if the phy has RGMII/1000-Base_T, but does not have the enet1g compatible, then an error is raised